test_delete_all_tasks.py

```diff
@@ -0,0 +1,145 @@
+
+def test_two_task_list(two_task_list, capsys):
+    #arrange / act
+    two_task_list.delete_all_tasks()
+
+    # assert
+    captured = capsys.readouterr()
+    assert captured.out == "All tasks deleted.\n" # console message to be printed
+    assert two_task_list.tasks == [] # length of 0s
+
+def test_delete_twice_empty(no_tasks_list, capsys):
+    # arrange / act
+    no_tasks_list.delete_all_tasks() # delete tasks from empty list once
+    
+    captured = capsys.readouterr() # ensure delete was successful
+    assert captured.out == "No tasks to delete.\n"
+    assert no_tasks_list.tasks == []
+
+    no_tasks_list.delete_all_tasks() # now delete twice
+    
+    
+    assert captured.out == "No tasks to delete.\n" # ensure the second delete also worked as expected 
+    assert no_tasks_list.tasks == []
+
+def test_delete_twice_one(one_task_list, capsys):
+    # arrange / act
+    one_task_list.delete_all_tasks()
+    # check output for correct 
+    captured = capsys.readouterr()
+    assert captured.out == "All tasks deleted.\n"
+    assert one_task_list.tasks == []
+
+    # rerun deleting and checking 
+    one_task_list.delete_all_tasks()
+    captured = capsys.readouterr()
+    assert captured.out == "No tasks to delete.\n"
+    assert one_task_list.tasks == []
+
+def test_delete_twice_two(two_task_list, capsys):
+    # arrange / act
+    two_task_list.delete_all_tasks()
+    # check first deletion is acting correctly
+    captured = capsys.readouterr()
+    assert captured.out == "All tasks deleted.\n"
+    assert two_task_list.tasks == []
+    # now testing all facets of the second
+    two_task_list.delete_all_tasks()
+    captured = capsys.readouterr()
+    assert captured.out == "No tasks to delete.\n"
+    assert two_task_list.tasks == []
+
+
+def test_delete_add_delete_one(homework_task, one_task_list):
+    one_task_list.delete_all_tasks()
+    one_task_list.add_task(homework_task)
+    one_task_list.delete_all_tasks()
+
+
+    
+    assert one_task_list.tasks == []
+
+def test_delete_add_delete_two(homework_task, two_task_list):
+    two_task_list.delete_all_tasks()
+    two_task_list.add_task(homework_task)
+    two_task_list.delete_all_tasks()
+
+    assert two_task_list.tasks == []
+
+def test_delete_add_delete_empty(homework_task, no_tasks_list, capsys):
+    no_tasks_list.delete_all_tasks()
+    no_tasks_list.add_task(homework_task)
+    no_tasks_list.delete_all_tasks()
+
+    assert no_tasks_list.tasks == []
+
+def test_large_task_list_deletion(homework_task):
+    todo_list = TodoList()
+    for _ in range(1000):  # Add the same homework task multiple times
+        todo_list.add_task(homework_task)
+    todo_list.delete_all_tasks()
+    assert todo_list.tasks == []
+
+def test_large_task_empty_deletion(empty_task):
+    todo_list = TodoList()
+    for _ in range(1000):  # Add the same homework task multiple times
+        todo_list.add_task(empty_task)
```

Comment:
Similarly, this method won't add a task if a duplicate task already exists and is testing deleting all tasks from a list with a single task, not with 1000.

Reply:
changed this method in most recent commit